node-zerox/src/utils.ts

```diff
@@ -11,9 +11,18 @@ export const encodeImageToBase64 = async (imagePath: string) => {
 
 // Strip out the ```markdown wrapper
 export const formatMarkdown = (text: string) => {
-  const formattedMarkdown = text
-    .replace(/^```[a-z]*\n([\s\S]*?)\n```$/g, "$1")
-    .replace(/^```\n([\s\S]*?)\n```$/g, "$1");
+  let formattedMarkdown = text;
+  let loopCount = 0;
+  const maxLoops = 3;
+
+  while (formattedMarkdown.startsWith("```markdown") && loopCount < maxLoops) {
+    formattedMarkdown = formattedMarkdown
```

Comment:
This is a great example for testing:

const text = `\`\`\`markdown
\`\`\`markdown
## Subtitle

Some text with \`inline code\`.

\`\`\`python
def hello_world():
    print("Hello, World!")
\`\`\`

More text here.
\`\`\`
\`\`\`
`;

For me, it seems to remove the first `\`\`\  instance that it finds. So in the end we have,
## Subtitle

Some text with `inline code`.

```python
def hello_world():
    print("Hello, World!")

More text here.
```

Reply:
Ahh, it does not come through correctly here.

I think we need to check if it starts with "`\`\`\markdown" and ends with "`\`\`\"

Reply:
Might need something like this:

```
const formatMarkdown = (text) => {
  let formattedMarkdown = text.trim();
  let loopCount = 0;
  const maxLoops = 3;

  const startsWithMarkdown = formattedMarkdown.startsWith("```markdown");
  while (startsWithMarkdown && loopCount < maxLoops) {
    const endsWithClosing = formattedMarkdown.endsWith("```");

    if (startsWithMarkdown && endsWithClosing) {
      const outermostBlockRegex = /^```markdown\n([\s\S]*?)\n```$/;
      const match = outermostBlockRegex.exec(formattedMarkdown);

      if (match) {
        formattedMarkdown = match[1].trim();
        loopCount++;
      } else {
        break;
      }
    } else {
      break;
    }
  }

  return formattedMarkdown;
};
```